charts/iter8lib/templates/_task-grpc.tpl

```diff
@@ -1,7 +1,45 @@
 {{- define "task.grpc" -}}
-# task: generate gRPC requests for the given gRPC method
+{{/* Validate values */}}
+{{- if not .Values.host }}
+{{- fail "Please set a value for the host parameter." }}
+{{- end }}
+{{- if not .Values.call }}
+{{- fail "Please set a value for the call parameter." }}
+{{- end }}
+{{/* Perform the various setup steps before the main task */}}
+{{- $vals := mustDeepCopy .Values }}
+{{- if .Values.protoURL }}
+# task: download proto file from URL
+- run: |
+    curl -o ghz.proto {{ $vals.protoURL }}
+{{- $pf := dict "proto" "ghz.proto" }}
+{{- $vals = mustMerge $pf .Values }}
+{{- end }}
+{{- if .Values.dataURL }}
+# task: download JSON data file from URL
+- run: |
+    curl -o data.json {{ $vals.dataURL }}
```

Comment:
It is my impression that the purpose of this (and similar curl commands) here and in `_task-http.tpl` is to avoid worrying about certificates. I am not sure I see how using curl avoids this problem. To be more bulletproof, we might include `-k` and `-L` as options.